src/sources/apache_metrics/parser.rs

```diff
@@ -0,0 +1,837 @@
+            }),
+        },
+        "ConnsAsyncClosing" => match value.parse::<f64>() {
+            Ok(value) => LineResult::Metrics(vec![Metric {
+                name: encode_namespace(namespace, "connections"),
+                timestamp: Some(now),
+                tags: {
+                    let mut tags = tags.map(|tags| tags.clone()).unwrap_or_default();
+                    tags.insert("state".to_string(), "closing".to_string());
+                    Some(tags)
+                },
+                kind: MetricKind::Absolute,
+                value: MetricValue::Gauge { value },
+            }]),
+            Err(err) => LineResult::Error(ParseError {
+                key: key.to_string(),
+                err: err.into(),
+            }),
+        },
+        "ConnsAsyncKeepAlive" => match value.parse::<f64>() {
+            Ok(value) => LineResult::Metrics(vec![Metric {
+                name: encode_namespace(namespace, "connections"),
+                timestamp: Some(now),
+                tags: {
+                    let mut tags = tags.map(|tags| tags.clone()).unwrap_or_default();
+                    tags.insert("state".to_string(), "keepalive".to_string());
+                    Some(tags)
+                },
+                kind: MetricKind::Absolute,
+                value: MetricValue::Gauge { value },
+            }]),
+            Err(err) => LineResult::Error(ParseError {
+                key: key.to_string(),
+                err: err.into(),
+            }),
+        },
+        "Scoreboard" => {
+            let to_metric = |state: &str, count: &u32| Metric {
+                name: encode_namespace(namespace, "scoreboard"),
+                timestamp: Some(now),
+                tags: {
+                    let mut tags = tags.map(|tags| tags.clone()).unwrap_or_default();
+                    tags.insert("state".to_string(), state.to_string());
+                    Some(tags)
+                },
+                kind: MetricKind::Absolute,
+                value: MetricValue::Gauge {
+                    value: (*count).into(),
+                },
+            };
+
+            let scores = value.chars().fold(HashMap::new(), |mut m, c| {
+                *m.entry(c).or_insert(0u32) += 1;
+                m
+            });
+
+            let scoreboard: HashMap<char, &str> = vec![
+                ('_', "waiting"),
+                ('S', "starting"),
+                ('R', "reading"),
+                ('W', "sending"),
+                ('K', "keepalive"),
+                ('D', "dnslookup"),
+                ('C', "closing"),
+                ('L', "logging"),
+                ('G', "finishing"),
+                ('I', "idle_cleanup"),
+                ('.', "open"),
+            ]
+            .into_iter()
+            .collect();
+
+            LineResult::Metrics(
+                scoreboard
+                    .iter()
+                    .map(|(c, name)| to_metric(name, scores.get(c).unwrap_or(&0u32)))
```

Comment:
`unwrap_or_default()` should work here as well I believe.

Reply:
Although that would probably require something similar to `(*scores.get(c)).unwrap_or_default()` to dereference the `u32`.

Reply:
:+1: I ended up with `scores.get(c).copied().unwrap_or_default()`.